go/test/endtoend/vreplication/migrate_test.go

```diff
@@ -0,0 +1,163 @@
+		execVtgateQuery(t, conn, "rating:0", "insert into rating(gid, pid, rating) values(2, 2, 5);")
+	})
+}
+
+// TestMigrate runs an e2e test for importing from an external cluster using the Mount and Migrate commands.
+// We have an anti-pattern in Vitess: vt executables look for an environment variable VTDATAROOT for certain cluster parameters
+// like the log directory when they are created. Until this test we just needed a single cluster for e2e tests.
+// However now we need to create an external Vitess cluster. For this we need a different VTDATAROOT and
+// hence the VTDATAROOT env variable gets overwritten.
+// Each time we need to create vt processes in the "other" cluster we need to set the appropriate VTDATAROOT
+func TestMigrate(t *testing.T) {
+	defaultCellName := "zone1"
+	cells := []string{"zone1"}
+	allCellNames = "zone1"
+	vc = NewVitessCluster(t, "TestMigrate", cells, mainClusterConfig)
+
+	require.NotNil(t, vc)
+	defaultReplicas = 0
+	defaultRdonly = 0
+	//defer vc.TearDown()
+
+	defaultCell = vc.Cells[defaultCellName]
+	vc.AddKeyspace(t, []*Cell{defaultCell}, "product", "0", initialProductVSchema, initialProductSchema, defaultReplicas, defaultRdonly, 100)
+	vtgate = defaultCell.Vtgates[0]
+	require.NotNil(t, vtgate)
+	vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.master", "product", "0"), 1)
+
+	vtgateConn = getConnection(t, vc.ClusterConfig.hostname, vc.ClusterConfig.vtgateMySQLPort)
+	defer vtgateConn.Close()
+	verifyClusterHealth(t, vc)
+	insertInitialData(t)
+
+	// create external cluster
+	extCell := "extcell1"
+	extCells := []string{extCell}
+	extVc := NewVitessCluster(t, "TestMigrateExternal", extCells, externalClusterConfig)
+	require.NotNil(t, extVc)
+	//defer extVc.TearDown()
+
+	extCell2 := extVc.Cells[extCell]
+	extVc.AddKeyspace(t, []*Cell{extCell2}, "rating", "0", initialExternalVSchema, initialExternalSchema, 0, 0, 1000)
+	extVtgate := extCell2.Vtgates[0]
+	require.NotNil(t, extVtgate)
+
+	extVtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.master", "rating", "0"), 1)
+	verifyClusterHealth(t, extVc)
+	extVtgateConn := getConnection(t, extVc.ClusterConfig.hostname, extVc.ClusterConfig.vtgateMySQLPort)
+	insertInitialDataIntoExternalCluster(t, extVtgateConn)
+
+	var err error
+	var output, expected string
+	ksWorkflow := "product.e1"
+
+	t.Run("mount external cluster", func(t *testing.T) {
+		if output, err = vc.VtctlClient.ExecuteCommandWithOutput("Mount", "-type=vitess", "-topo_type=etcd2",
+			fmt.Sprintf("-topo_server=localhost:%d", extVc.ClusterConfig.topoPort), "-topo_root=/vitess/global", "ext1"); err != nil {
+			t.Fatalf("Mount command failed with %+v : %s\n", err, output)
+		}
+		if output, err = vc.VtctlClient.ExecuteCommandWithOutput("Mount", "-type=vitess", "-list"); err != nil {
+			t.Fatalf("Mount command failed with %+v : %s\n", err, output)
+		}
+		expected = "ext1\n"
+		require.Equal(t, expected, output)
```

Comment:
suggestion: `require.Contains(t, "exct1", output)`, so that we don't need to worry about `"\n"`. Of course, your test is stricter.

Reply:
Explicitly added the stricter test because of a previous case where a \n had not been missed, while listing workflows.

Reply:
Good to know!